### PR TITLE
Feature/ml model improved accuracy

### LINKED ISSUE #21: Enhancing the accuracy of crop yield prediction model
The yield prediction model right now already performs pretty good.
But its accuracy can be improved more using other ensemble techniques like catboost, stacking and using ridge regularization to avoid risk of overfit.
Also the the Pearson correlation check was conducted on the whole dataset which triggers the risk of data leakage.
These steps can ensure an even better statistical accuracy.

SIR, PLEASE ASSIGN THIS ISSUE TO ME.

THANK YOU!

### PR DESCRIPTION
**Pull request by sagnik17299**
1)Improved model accuracy by using Stack Ensemble technique on XGBOOST AND CATBOOST models which are specifically good for categorical data. 
2)Accuracy improved from 0.93 to 0.94 R2 score and Root mean squared error decreased.
3)Used Pearson Correlation test on train data only instead of whole dataset to avoid data leakage and bias.
Fixes #21

### CODE REVIEW

**Comment by omroy07** [comment]
> Please check and merge it if you find it okay. Let me know if any correction is needed. Thank you @omroy07

Okay
